cmd/monitor/monitor.go

```diff
@@ -0,0 +1,201 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/skycoin/skycoin/src/util/logging"
+)
+
+var (
+	logger = logging.MustGetLogger("monitor")
+	// ErrPeerlistFull is returned when the Pex is at a maximum
+	ErrPeerlistFull = errors.New("Peer list full")
+	// ErrInvalidAddress is returned when an address appears malformed
+	ErrInvalidAddress = errors.New("Invalid address")
+	// ErrNoLocalhost is returned if a localhost addresses are not allowed
+	ErrNoLocalhost = errors.New("Localhost address is not allowed")
+	// ErrNotExternalIP is returned if an IP address is not a global unicast address
+	ErrNotExternalIP = errors.New("IP is not a valid external IP")
+	// ErrPortTooLow is returned if a port is less than 1024
+	ErrPortTooLow = errors.New("Port must be >= 1024")
+	// ErrBlacklistedAddress returned when attempting to add a blacklisted peer
+	ErrBlacklistedAddress = errors.New("Blacklisted address")
+
+	whitespaceFilter = regexp.MustCompile(`\s`)
+
+	dialTimeout = 5 * time.Second
+	urlPeers    = os.Getenv("GOPATH") + "/src/github.com/skycoin/skycoin/peers.txt"
+)
+
+// validateAddress returns a sanitized address if valid, otherwise an error
+func validateAddress(ipPort string) (string, error) {
```

Comment:
if this function is already defined elsewhere in `src/` isn't it better to export it for reuse ?

```
$ grep -nr 'func validateAddress' src/
src/daemon/pex/pex.go:64:func validateAddress(ipPort string, allowLocalhost bool) (string, error) {
```